packages/eslint-plugin-ckeditor5-rules/lib/rules/ckeditor-plugin-flags.js

```diff
@@ -0,0 +1,352 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const PLUGIN_NAME_METHOD_NAME = 'pluginName';
+
+module.exports = {
+	meta: {
+		type: 'problem',
+		docs: {
+			description: 'Enforce the presence of the plugin flags.',
+			category: 'CKEditor5'
+		},
+		schema: [
+			{
+				type: 'object',
+				properties: {
+					requiredFlags: {
+						type: 'array',
+						items: {
+							type: 'object',
+							properties: {
+								name: { type: 'string' },
+								returnValue: { type: 'boolean' }
+							},
+							required: [ 'name', 'returnValue' ]
+						},
+						optional: true
+					},
+					disallowedFlags: {
+						type: 'array',
+						items: {
+							type: 'string'
+						},
+						optional: true
+					}
+				},
+				additionalProperties: false
+			}
+		],
+		messages: {
+			disallowedFlag: 'The class member \'{{flag}}\' is disallowed here and should be removed.',
+			missingRequiredFlag: 'The class should have a static getter method \'{{flag}}\' that returns {{returnValue}}.',
+			incorrectReturnValue: 'The method \'{{flag}}\' should return {{returnValue}} but returns {{actualValue}} instead.',
+			notPublicGetter: 'The method \'{{flag}}\' should be defined as a public getter method.'
+		}
+	},
+
+	create( context ) {
+		if ( !getRequiredFlagsConfig( context ).length && !getDisallowedFlagsConfig( context ).length ) {
+			return {};
+		}
+
+		return {
+			'ClassDeclaration:exit'( classDeclaration ) {
+				if ( isPluginClassDeclaration( classDeclaration ) ) {
+					checkIfThereAreNoDisallowedFlags( context, classDeclaration );
+					checkIfAllFlagsAreProperlyDefined( context, classDeclaration );
+				}
+			}
+		};
+	}
+};
+
+/**
+ * Returns the required flags from the rule options.
+ *
+ * @param {Object} context The ESLint rule context.
+ * @returns {Array.<Object>}
+ */
+function getRequiredFlagsConfig( context ) {
+	return context.options[ 0 ]?.requiredFlags ?? [];
+}
+
+/**
+ * Returns the disallowed flags from the rule options.
+ *
+ * @param {Object} context The ESLint rule context.
+ * @returns {Array.<String>}
+ */
+function getDisallowedFlagsConfig( context ) {
+	return context.options[ 0 ]?.disallowedFlags ?? [];
+}
+
+/**
+ * Checks if there are no disallowed flags defined in the class declaration.
+ *
+ * @param {Object} context The ESLint rule context.
+ * @param {Object} classDeclaration The class declaration node to analyze.
+ * @example
+ *
+ * Let's assume we have the following class declaration:
+ *
+ * ```
+ * class TestPlugin extends Plugin {
+ * 	static get pluginName() { return 'TestPlugin'; }
+ * 	static get isOfficialPlugin() { return true; }
```

Comment:
It's still possible to override this rule using casting to `any`. I'm wondering if that's going to be a problem or not 🤔
![image](https://github.com/user-attachments/assets/60ae6748-7939-41e4-a0ac-15d4c5dfb94d)

Reply:
We decided to keep this rule as simple as it's possible, without any further `return` checks, order checks, etc. So while we can, and it'll be not detected, it's still fine for scope of this PR. 

In fact, more sophisticated checks are present in removed commit, right here: https://github.com/ckeditor/ckeditor5-linters-config/pull/45/commits/8173888ecdc7db29198796491ba4ef82b328970e#diff-e3282f92e404601c5ff51bbb33f3a3a7d3ce098c20cede15ad52b2fa0c8e35cfL247-L278
So there's still a way to revert it (just in case).